ballerina/producer.bal

```diff
@@ -116,7 +116,7 @@ public client isolated class Producer {
         } else if anydataKey !is () {
             key = anydataKey.toJsonString().toBytes();
         }
-        return sendByteArrayValues(self, value, producerRecord.topic, key,
+        return sendByteArrayValues(self, value, producerRecord.topic, producerRecord.headers, key,
```

Comment:
Can we use the optional field access here. Eventhough both options work for this scenario, I thought this will be more readable and indicate we are going to have a parameter which can be `nil` if not present

```suggestion
        return sendByteArrayValues(self, value, producerRecord.topic, producerRecord?.headers, key,
```

Reply:
Will update